components/api/src/main/java/com/hotels/styx/api/StyxCoreObservable.java

```diff
@@ -103,4 +83,21 @@ public StyxCoreObservable(CompletionStage<T> future) {
         return future;
     }
 
+    private static <U> Observable<? extends U> toObservable(StyxObservable<U> styxObservable) {
```

Comment:
Do you see any advantage to the return type? It's usually considered a bad practice to include wildcard types in return types as, rather than providing more flexibility to the caller, it's just makes the code more bloated.

If we pass a StyxObservable<U> we are always going to return an Observable<U> and not any subclass of U.

 BTW,  T seems better than U if we want to be consistent.

Reply:
I didn't know that was in the Java tutorials! I  thought it was usually quoted by the 'Effective Java' book.

Reply:
This one is easily fixed as it is private and doesn't break anything. As for which letters to use for type parameters, both Rx and java streams use `T` for input and `R` for output.